test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -395,12 +395,14 @@ protected void registerOutboundFilter(ICluster cluster)
         MessagingService.instance().outboundSink.add((message, to) -> {
             if (isShutdown())
                 return false; // TODO: Simulator needs this to trigger a failure
-            IMessage serialzied = serializeMessage(message.from(), to, message);
             int fromNum = config.num(); // since this instance is sending the message, from will always be this instance
             IInstance toInstance = cluster.get(fromCassandraInetAddressAndPort(to));
             if (toInstance == null)
                 return true; // TODO: Simulator needs this to trigger a failure
+            if (!cluster.filters().hasInbound() && !cluster.filters().hasOutbound())
```

Comment:
perf issue, tests would have no filtering yet we would spend a ton of time deserializing to check.